src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -121,16 +121,15 @@ private MsQuicApi(NativeApi* vtable)
 
         static MsQuicApi()
         {
-            // TODO: Consider updating all of these delegates to instead use function pointers.
             if (NativeLibrary.TryLoad(Interop.Libraries.MsQuic, typeof(MsQuicApi).Assembly, DllImportSearchPath.AssemblyDirectory, out IntPtr msQuicHandle))
             {
                 try
                 {
-                    if (NativeLibrary.TryGetExport(msQuicHandle, "MsQuicOpen", out IntPtr msQuicOpenAddress))
+                    if (NativeLibrary.TryGetExport(msQuicHandle, "MsQuicOpenVersion", out IntPtr msQuicOpenVersionAddress))
                     {
-                        MsQuicOpenDelegate msQuicOpen =
-                            Marshal.GetDelegateForFunctionPointer<MsQuicOpenDelegate>(msQuicOpenAddress);
-                        uint status = msQuicOpen(out NativeApi* vtable);
+                        delegate* unmanaged[Cdecl]<uint, out NativeApi*, uint> msQuicOpenVersion =
+                            (delegate* unmanaged[Cdecl]<uint, out NativeApi*, uint>)msQuicOpenVersionAddress;
+                        uint status = msQuicOpenVersion(1, out NativeApi* vtable);
```

Comment:
Is the 1 version number? If so perhaps something like const int MsQuicVersion = 1 would be good. 
How does it correspond to MsQuic version e.g. 1.15?

Reply:
Yeah I can make a constant. And yes it will match the major version.

Reply:
If you want to do something like restrict to >= 1.4, you open with 1, query the `QUIC_PARAM_GLOBAL_VERSION` and then check the minor (`globalVersion[1]`) to make sure it's `>= 4`. This would then allow you to update the msquic binary independently from .NET, and still be version safe.

Reply:
ok. I'm not trying to solve particular problem. This is mostly about support and diagnostic. 
Having said that, there seems to be APIs in 1.5 that are not In 1.4, right? How would this help? 
Or is the plan to rev the version now more frequently? (e.g. on each API change?)

Reply:
This doesn't help for non breaking changes, unless you do the minor version check. We still plan on only updating the version if a breaking change occurs.

Reply:
How will that work for newly built msquic dll? It will load old API methods but will include new changes, if they were made to existing methods? Or if the version is already 2, loading will fail?

Reply:
Newer versions of the library will support the older API table. So if you request API version 1, you will get version 1. If you specifically need something like 1.6 or later, you would load API version 1, and then you can call GetParam with version parameters to get the minor version number of the binary. Since we can't break that API without bumping the version table, that is guaranteed to work.